src/renderer/components/Login.js

```diff
@@ -15,32 +16,63 @@ const {
 class Login extends React.Component {
   constructor (props) {
     super(props)
-    this.state = {
-      addr: process.env.DC_ADDR,
-      showAdvanced: false,
-      mailUser: null,
-      mailPw: process.env.DC_MAIL_PW,
-      mailServer: null,
-      mailPort: null,
-      mailSecurity: null,
-      sendUser: null,
-      sendPw: null,
-      sendServer: null,
-      sendPort: null,
-      sendSecurity: null
-    }
+    this.state = this._defaultState(false)
 
     this.handleChange = this.handleChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
     this.handleShowAdvanced = this.handleShowAdvanced.bind(this)
   }
 
+  _defaultState (showAdvanced) {
+    return {
+      addr: process.env.DC_ADDR ? process.env.DC_ADDR : '',
+      mailUser: '',
+      mailPw: process.env.DC_MAIL_PW ? process.env.DC_MAIL_PW : '',
+      mailServer: '',
+      mailPort: '',
+      mailSecurity: '',
+      sendUser: '',
+      sendPw: '',
+      sendServer: '',
+      sendPort: '',
+      sendSecurity: '',
+      showAdvanced: showAdvanced
+    }
+  }
+
   handleChange (event) {
     var state = {}
     state[event.target.id] = event.target.value
     this.setState(state)
   }
 
+  translateSecurityToServerFlags (mailSecurity, sendSecurity) {
+    let serverFlags = []
+    if (mailSecurity) {
+      let imapFlag
+      if (mailSecurity === 'ssl') {
```

Comment:
perhaps we could create a constant dictionary and look up the value in the dictionary instead of having `if` statements.

Reply:
May be better, will check it out.